src/main/org/deidentifier/arx/algorithm/DataDependentEDDPAlgorithm.java

```diff
@@ -183,12 +171,15 @@ private long executeExponentialMechanism(Map<Long, ILScore> transformationIDToSc
         // Convert the map into arrays of the types required by the exponential mechanism
 
         Long[] values = new Long[transformationIDToScore.size()];
-        BigFraction[] scores = new BigFraction[values.length];
+        double[] scores = new double[values.length];
         
         int i = 0;
         for (Entry<Long, ILScore> entry : transformationIDToScore.entrySet()) {
             values[i] = entry.getKey();
-            scores[i] = entry.getValue().getValue();
+            scores[i] = entry.getValue().getValue().doubleValue();
+            if (Double.isInfinite(scores[i]) || Double.isNaN(scores[i])) {

```

Comment:
How can you be sure that no rounding is involved in the conversion to double? I can find nothing in that regard in the documentation. How do you know that NaN or +/-Infinity can be produced?